app/controllers/redirections_controller.rb

```diff
@@ -0,0 +1,13 @@
+class RedirectionsController < ApplicationController
+  def next
+    redirection = Redirection.find_by(slug: params[:slug])
+
+    redirect_to redirection.next.url
+  end
+
+  def previous
+    redirection = Redirection.find_by(slug: params[:slug])
+
+    redirect_to redirection.previous.url
```

Comment:
What about creating a method on `Redirection` to avoid this Demeter violation?

```ruby
# redirection.rb
# ...
  def previous_url
    previous.url
  end
```